src/main/java/io/vavr/collection/Set.java

```diff
@@ -163,6 +163,15 @@ default Boolean apply(T element) {
      */
     Set<T> removeAll(Iterable<? extends T> elements);
 
+    /**
+     * Reverse the order of all given elements in this set, if present.
+     *
+     * @return A new set consisting of the elements in reverse order of this set
+     */
+    default Set<T> reverse() {
+        throw new UnsupportedOperationException("Operation not supported");
+    }
```

Comment:
This makes are life harder than it needs to be. A Set is unordered by definition. Please remove this method from Set and add it to SortedSet, but with no default implementation. We need to duplicate it then to LinkedHashSet but that's ok, it is the best we can do.

Could you please add it also to SortedMap and LinkedHashMap? We are then able to close the original issue.

Reply:
> This makes are life harder than it needs to be. A Set is unordered by definition. Please remove this method from Set and add it to SortedSet, but with no default implementation. We need to duplicate it then to LinkedHashSet but that's ok, it is the best we can do.
> 

Thank you for pointing out my mistake. Will make the suggested changes. But I have one concern regarding`SortedSet`. It is also implemented by `BitSet`. How are we going to handle it. Should I throw `UnsupportedOperartionException` in `BitSet#reverse()` or you have any other suggestion?